### ISSUE TITLE
AMP auto redirect breaks website

### ISSUE THREAD

**Issue by 0mMdy8R6uS8xpP** [brave/brave-browser#41665]
### Description

Brave browsers AMP auto redirect feature broke a website

### Steps to reproduce

1. Turn on brave AMP auto redirect
2. Go to https://m.nocutnews.co.kr/news/amp/6227673
3. The url becomes https://m.nocutnews.co.kr/news/&lt;%= resulting in broken site

### Actual result

I went to a google AMP page and brave browser broke the website, turning off AMP auto redirect feature fixes this

### Expected result

Brave browser should redirect to https://m.nocutnews.co.kr/news/6227673 instead of https://m.nocutnews.co.kr/news/&lt;%= when user tries to access https://m.nocutnews.co.kr/news/amp/6227673

### Reproduces how often

Easily reproduced

### Desktop Brave version (brave://version info)

v1.70.126 on Windows 11

### Android device

- Brand/model: Galaxy A23
- Android version: 14

### Channel information

- [x] release (stable)
- [x] beta
- [x] nightly

### Reproducibility

- [x] with Brave Shields disabled
- [x] with Brave Rewards disabled
- [ ] in the latest version of Chrome

**Comment by ShivanKaul** [comment]
The AMP canonical link element is malformed for the website: `<link rel="canonical" href="https://m.nocutnews.co.kr/news/<%= Index %>">`
Given that the website does not follow the AMP spec, I'm inclined to call this a wontfix. Thoughts @boocmp?

**Comment by boocmp** [comment]
Yes it is. It references ```<link rel="canonical" href="https://m.nocutnews.co.kr/news/&lt;%= Index %">```.

**Comment by ShivanKaul** [comment]
I wonder if we could do something clever here: if the target URL after a de-amp results in an error status code (`4xx` or `5xx`), then we fallback to the original URL. WDYT?

**Comment by ShivanKaul** [comment]
@boocmp pointed out that in this case we should just have a check that the regex matches the href, and that would've resulted in the existing fallback.

**Comment by ShivanKaul** [comment]
additionally we should also do https://github.com/brave/brave-browser/issues/41665#issuecomment-2418275201
